core/jmx/impl/src/main/java/org/opennms/netmgt/jmx/impl/connection/connectors/Jsr160ConnectionFactory.java

```diff
@@ -72,21 +83,36 @@ public static JmxServerConnectionWrapper getMBeanServerConnection(Map<String,?>
         String urlPath  = ParameterMap.getKeyedString( propertiesMap, "urlPath",  "/jmxrmi");
         String username = ParameterMap.getKeyedString(propertiesMap, "username", null);
         String password = ParameterMap.getKeyedString(propertiesMap, "password", null);
+        long timeout     = ParameterMap.getKeyedLong(propertiesMap, "timeout", DEFAULT_TIMEOUT);
         
-        Jsr160ConnectionWrapper connectionWrapper = null;
-        JMXServiceURL url = null;
+        //Jsr160ConnectionWrapper connectionWrapper = null;
         
         final String hostAddress = InetAddressUtils.str(address);
         LOG.debug("JMX: {} - service:{}//{}:{}{}", factory, protocol, hostAddress, port, urlPath);
 
         if (factory == null || factory.equals("STANDARD")) {
-                url = getUrl(address, port, protocol, urlPath);
+                final JMXServiceURL url = getUrl(address, port, protocol, urlPath);
                 
                 // Connect a JSR 160 JMXConnector to the server side
-                JMXConnector connector = JMXConnectorFactory.connect(url);
-                MBeanServerConnection connection = connector.getMBeanServerConnection();
-                
-                return new Jsr160ConnectionWrapper(connector, connection);
+                ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
I would make this executor static and final instead of re-creating it on every call.